x/supplier/client/cli/helpers_test.go

```diff
@@ -32,3 +54,197 @@ func networkWithSupplierObjects(t *testing.T, n int) (*network.Network, []shared
+		for _, appAcct := range appAccts {
+			for _, supplierAcct := range supplierAccts {
+				claim := createClaim(
+					t, net, ctx,
+					supplierAcct.Address.String(),
+					sessionEndHeight,
+					appAcct.Address.String(),
+				)
+				claims = append(claims, *claim)
+				// TODO_TECHDEBT(#196): Move this outside of the forloop so that the test iteration is faster
+				require.NoError(t, net.WaitForNextBlock())
+			}
+		}
+	}
+
+	return net, claims
+}
+
+func encodeSessionHeader(
+	t *testing.T,
+	appAddr string,
+	sessionId string,
+	sessionStartHeight int64,
+) string {
+	t.Helper()
+
+	argSessionHeader := &types2.SessionHeader{
+		ApplicationAddress:      appAddr,
+		SessionStartBlockHeight: sessionStartHeight,
+		SessionId:               sessionId,
+		SessionEndBlockHeight:   sessionStartHeight + numBlocksPerSession,
+		Service:                 &sharedtypes.Service{Id: testServiceId},
+	}
+	cdc := codec.NewProtoCodec(types3.NewInterfaceRegistry())
+	sessionHeaderBz := cdc.MustMarshalJSON(argSessionHeader)
+	return base64.StdEncoding.EncodeToString(sessionHeaderBz)
+}
+
+func createClaim(
+	t *testing.T,
+	net *network.Network,
+	ctx client.Context,
+	supplierAddr string,
+	sessionEndHeight int64,
+	appAddress string,
+) *types.Claim {
+	t.Helper()
+
+	rootHash := []byte("root_hash")
+	sessionStartHeight := sessionEndHeight - numBlocksPerSession
+	sessionId := getSessionId(t, net, appAddress, supplierAddr, sessionStartHeight)
+	sessionHeaderEncoded := encodeSessionHeader(t, appAddress, sessionId, sessionStartHeight)
+	rootHashEncoded := base64.StdEncoding.EncodeToString(rootHash)
+
+	args := []string{
+		sessionHeaderEncoded,
+		rootHashEncoded,
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, supplierAddr),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastSync),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, types4.NewCoins(types4.NewCoin(net.Config.BondDenom, math.NewInt(10))).String()),
+	}
+
+	responseRaw, err := cli.ExecTestCLICmd(ctx, cli2.CmdCreateClaim(), args)
+	require.NoError(t, err)
+	var responseJson map[string]interface{}
+	err = json.Unmarshal(responseRaw.Bytes(), &responseJson)
+	require.NoError(t, err)
+	require.Equal(t, float64(0), responseJson["code"], "code is not 0 in the response: %v", responseJson)
+
+	return &types.Claim{
+		SupplierAddress:       supplierAddr,
+		SessionId:             sessionId,
+		SessionEndBlockHeight: uint64(sessionEndHeight),
+		RootHash:              rootHash,
+	}
+}
+
+func getSessionId(
+	t *testing.T,
+	net *network.Network,
+	appAddr string,
+	supplierAddr string,
+	sessionStartHeight int64,
+) string {
+	t.Helper()
+	ctx := context.TODO()
+
+	sessionQueryClient := types2.NewQueryClient(net.Validators[0].ClientCtx)
+	res, err := sessionQueryClient.GetSession(ctx, &types2.QueryGetSessionRequest{
+		ApplicationAddress: appAddr,
+		Service:            &sharedtypes.Service{Id: testServiceId},
```

Comment:
Thoughts on having the caller pass in this local constant?

Reply:
Definitely have thoughts :sweat_smile:. This is related to https://github.com/pokt-network/poktroll/pull/236#discussion_r1415261843. 

TL;DR, I would  hold off for now.

It might be better to take a step back and think about how we would like multi-service integration tests to be designed, and then evaluate if it's compatible with our current direction (i.e. refactor and/or reuse what we have) or if we need to do that in a new test suite with more comprehensive helpers. Wdyt?

Reply:
I think holding off and seeing what/when we need from multi-service integration makes sense.

Let's hold off 👍

Reply:
:pray: